packages/core/src/app/checkout/Checkout.tsx

```diff
@@ -286,7 +292,7 @@ class Checkout extends Component<
         }
 
         return (
-            <div className={classNames({ 'is-embedded': isEmbedded() })}>
+            <div className={classNames({ 'is-embedded': isEmbedded(), 'remove_checkout_step_numbers': isHidingStepNumbers })}>
```

Comment:
Initially, I changed the html structure as well, but then I realised I could control every UI element with pure CSS if there was a specific class name at the top level.